lang/cs/Org.Apache.REEF.Wake/Remote/Impl/StreamingLink.cs

```diff
@@ -197,22 +184,12 @@ public T Read()
                 Exceptions.Throw(new IllegalStateException("Link has been disposed."), Logger);
             }
 
-            string dataType = "";
-
-            dataType = await _reader.ReadStringAsync(token);
-
-            if (dataType == null)
-            {
-                return default(T);
-            }
-
             try
             {
-                T value = (T) _injector.ForkInjector().GetInstance(dataType);
-                await value.ReadAsync(_reader, token);
+                T value = await _streamingCodec.ReadAsync(_reader, token);
                 return value;
             }
-            catch (InjectionException)
+            catch (Exception e)
```

Comment:
Same as above: Log the exception and provide justification for using `default` here.